middleware/grpc/example_client_test.go

```diff
@@ -0,0 +1,62 @@
+package grpc_test
+
+import (
+	"context"
+
+	datadogMiddleware "github.com/coopnorge/go-datadog-lib/v2/middleware/grpc"
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/testing/testpb"
+	"google.golang.org/grpc"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func ExampleUnaryClientInterceptor() {
+	ctx := context.Background()
+
+	cc, err := grpc.NewClient(
+		"https://example.com",
+		grpc.WithUnaryInterceptor(datadogMiddleware.UnaryClientInterceptor()),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer cc.Close()
+
+	client := testpb.NewTestServiceClient(cc)
+
+	span, ctx := tracer.StartSpanFromContext(ctx, "grpc.request")
+	resp, err := client.Ping(ctx, &testpb.PingRequest{})
+	span.Finish(tracer.WithError(err))
+	if err != nil {
+		panic(err)
+	}
+	println(resp)
+}
+
+func ExampleStreamClientInterceptor() {
+	ctx := context.Background()
+
+	cc, err := grpc.NewClient(
+		"https://example.com",
+		grpc.WithStreamInterceptor(datadogMiddleware.StreamClientInterceptor()),
+	)
+	if err != nil {
+		panic(err)
+	}
+	defer cc.Close()
+
+	client := testpb.NewTestServiceClient(cc)
+
+	span, ctx := tracer.StartSpanFromContext(ctx, "grpc.stream")
+	stream, err := client.PingStream(ctx)
+	defer span.Finish()
+	if err != nil {
+		span.Finish(tracer.WithError(err))
+		panic(err)
```

Comment:
Pretty sure this is OK, but noticed now that you have a defer span.FInish() already. Is it unproblematic to call span.Finish multiple times (assuming the deferred call will be executed after the panic)?

Reply:
According to https://go.dev/blog/defer-panic-and-recover is does. Normal code would probably not call panic, but return an error.